packages/common/client-utils/src/hashFileNode.ts

```diff
@@ -31,14 +32,17 @@ export async function hashFile(
 	// eslint-disable-next-line default-case
 	switch (algorithm) {
 		case "SHA-1": {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
```

Comment:
The package already dev-deps on `@types/sha.js`... could this be addressed by adding the types to tsconfig? Also, I looked at `sha.js`'s `index.js` and it seems to export the necessary things there (`sha1`, `sha256`). Maybe we can fix the imports at the top too?

Reply:
Unfortunately things don't work out as they aught to. We have to drill into `sha.js` with our imports to avoid transitively importing from the "crypto" library, which causes compilation issues. I didn't investigate much further into resolving those issues separately, but that's the reason for the drilling imports.

Since we are drilling, and the `@types/sha.js` library *only* contains type definitions for the package rollup (and doesn't have maps back to each individual source module), `TSC` doesn't know how to correlate the type definitions with the source 🫤

A more thorough dive _might_ be worth it at some point, but my main motivation here is to force us to make better choices with future changes, so I opted to leave it be for now.

Reply:
Thanks for looking into it!